### DISCUSSION TITLE
Do snaps support third party signers?

### DISCUSSION THREAD

**Discussion by anubhav6864** [MetaMask/snaps#1188] [Q&A]
Can we use a third party software or a hardware wallet in snaps to sign transactions and store keys?

**Comment by Montoya** [comment]
Hi @anubhav6864, you can create a snap that uses external signers. One limitation of the current snap API is that you cannot connect to USB devices, but if you have some other way of connecting to a hardware wallet then you can use the hardware wallet as a signer. 

If you are interested in building a snap that manages accounts in MetaMask, you should have a look at [Keyrings proposal](https://metamask.github.io/SIPs/SIPS/sip-2). This is not part of the Snaps API yet so it is not possible to use it in Flask but it is in development.
